File: Natixis Craftsmanship Practices/Breaking Dependencies/MarketTrading/java/src/main/java/com/markettrading/external/MarketChaos.java
package com.markettrading.external;

import java.util.Random;

@DoNotModify
public class MarketChaos {

    public long now() {
        return System.currentTimeMillis();
    }

    public int rnd(int bound) {
        return new Random().nextInt(bound);
    }
}

File: Natixis Craftsmanship Practices/Breaking Dependencies/MarketTrading/cpp/external/src/MarketChaos.cpp
#include "MarketChaos.h"

#include <cstdlib>
#include <ctime>

long MarketChaos::now() const {
    return std::time(nullptr) * 1000;
}

int MarketChaos::rnd(int bound) const {
    return std::rand() % bound;
}